go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -17,80 +17,115 @@ package main
 import (
 	"context"
 	"fmt"
+	"path/filepath"
 	"strings"
 )
 
 type BazelJSONBuilder struct {
-	bazel      *Bazel
-	query      string
-	tagFilters string
-	targets    []string
+	bazel    *Bazel
+	requests []string
 }
 
 const (
-	OutputGroupDriverJSONFile = "go_pkg_driver_json_file"
-	OutputGroupStdLibJSONFile = "go_pkg_driver_stdlib_json_file"
-	OutputGroupExportFile     = "go_pkg_driver_export_file"
+	RulesGoStdlibLabel = "@io_bazel_rules_go//:stdlib"
 )
 
-func NewBazelJSONBuilder(bazel *Bazel, query, tagFilters string, targets []string) (*BazelJSONBuilder, error) {
+func (b *BazelJSONBuilder) fileQuery(filename string) string {
+	if filepath.IsAbs(filename) {
+		fp, _ := filepath.Rel(b.bazel.WorkspaceRoot(), filename)
+		filename = fp
+	}
+	return fmt.Sprintf(`some(kind("go_library", same_pkg_direct_rdeps("%s")))`, filename)
+}
+
+func (b *BazelJSONBuilder) packageQuery(importPath string) string {
+	if strings.HasSuffix(importPath, "/...") {
+		importPath = strings.TrimSuffix(importPath, "/...") + "/.*"
+	}
+	return fmt.Sprintf(`some(kind("go_library", attr(importpath, "%s", deps(%s))))`, importPath, bazelQueryScope)
+}
+
+func (b *BazelJSONBuilder) queryFromRequests(requests ...string) string {
+	ret := make([]string, 0, len(requests))
+	for _, request := range requests {
+		if request == "." || request == "./..." {
+			if bazelQueryScope != "" {
+				ret = append(ret, fmt.Sprintf(`kind("go_library", %s)`, bazelQueryScope))
+			} else {
+				ret = append(ret, fmt.Sprintf(RulesGoStdlibLabel))
+			}
+		} else if request == "builtin" || request == "std" {
+			ret = append(ret, fmt.Sprintf(RulesGoStdlibLabel))
+		} else if strings.HasPrefix(request, "file=") {
+			f := strings.TrimPrefix(request, "file=")
+			ret = append(ret, b.fileQuery(f))
+		} else if bazelQueryScope != "" {
```

Comment:
is `bazelQueryScope` mandatory to accept package queries?

Reply:
i'd like it to, because else it's very easy to shoot yourself in the foot